### DISCUSSION TITLE
Integrating variable plot radii in plot_metrics()

### DISCUSSION THREAD

**Discussion by benice2me** [r-lidar/lidR#494] [Q&A] [answered]
Hi everyone,

For an ABA project I am currently working on I am trying to compute plot metrics using `plot_metrics`. The catch is that my plots have different radii (radiuses?). My plot_data, consisting of 165 rows and 16 columns, has a column containing these radii. What is the most efficient way to integrate these in the `plot_metrics` formula? 

Thanks in advance and kind regards,
Joep

Also: first time posting on GitHub, please excuse my lack of formatting.

**Comment by Jean-Romain** [comment, marked as answer]
You can provide several radii (one per plot)

```r
plot_metrics(ctg, centers, radius = center$radius)
```

**Reply by benice2me** [reply]
Thank you for the fast reply. Do you have a link to where this is documented? Also, is `centers` in your example the plot data?

**Reply by Jean-Romain** [reply]
> Also, is centers in your example the plot data?

`center` is your source of spatial points (e.g. a shapefile) in a `sp` or `sf` format. But you can do it another way

> Do you have a link to where this is documented?

Well, thechnically it is documented but... The doc states

>  `plot_metrics` is basically a seamless wrapper around `clip_roi`, `cloud_metrics`, `cbind` and adequate processing settings.

Then you look at `clip_roi`

> `...` in clip_roi: optional supplementary options (see supported geometries). Unused in other functions

So we look at *supported geometries*

> `Polygon`, `SpatialPolygons`, `SpatialPolygonsDataFrame`, `SpatialPoints` or `SpatialPointsDataFrame` in that case a parameter `radius` must be passed in `...`

Then we look at `radius` from `clip_circle`

> `radius`  numeric. disc radius or radii.

As a conclusion, and to be honnest, it is not documented...

**Reply by benice2me** [reply]
Thank you so much for answering my questions. Although I really should have looked into the `clip_roi` documentation, you might want to add a little footnote about this to the doc in due time :wink:

**Comment by benice2me** [comment]
I hate asking about this again but I am still unsure about the metrics which were calculated. I have tried the following 4 lines:
```

plot_metrics(catalog, .stdmetrics_z, plots, radius = plots$STRAAL) # the way you suggested, STRAAL is the header of my column containing radii
plot_metrics(catalog, .stdmetrics_z, plots, radius = 14) # 14 is the first radius in the list
plot_metrics(catalog, .stdmetrics_z, plots, radius = 10) # a somewhat arbitrary number
plot_metrics(catalog, .stdmetrics_z, plots, radius = list_r) # with list_r being a list of the radii
```
I have checked a couple of calculated metrics, such as zmax, zmean, zsd, zpcum7 and zpcum8, and they seem to be the exact same for every different radius input. Do you have an idea about how this is happening?

Sadly I cannot share my data with you.

**Comment by Jean-Romain** [comment]
This is not what I'm seeing with a reproducible example


```r
LASfile <- system.file("extdata", "Megaplot.laz", package="lidR")
las = readLAScatalog(LASfile)
col = sf::sf.colors(15)

shpfile <- system.file("extdata", "efi_plot.shp", package="lidR")
inventory <- sf::st_read(shpfile, quiet = TRUE)
inventory # contains an ID and a Value Of Interest (VOI) per plot
fun = ~list(q85 = quantile(Z, probs = 0.85))

m1 <- plot_metrics(las, fun, inventory, radius = 11.28)
m2 <- plot_metrics(las, fun, inventory, radius = runif(5, 5, 25))

plot(m1["q85"], pch = 19, cex = 3)
plot(m2["q85"], pch = 19, cex = 3)

m1$q85
#> [1] 21.4225 22.0655 23.0420 19.0825 14.9125
m2$q85
#> [1] 21.000 21.680 22.890 21.541 14.660
```
